src/app/components/resources/resources.component.ts

```diff
@@ -13,23 +13,26 @@ export class ResourcesComponent implements OnInit {
   constructor(private resourceService: ResourceService, private languageService: LanguageService) {}
 
   ngOnInit(): void {
-    this.resourceService.getResources().then(resources => this.resources = resources);
+    this.resourceService.getResources(null).then(resources => this.resources = resources);
   }
 
   loadTranslations(resource): void {
-    this.resourceService.getResource(resource.id)
-      .then((r) => {
-        resource.latest = r['latest-translations'];
+    let x = 0;
+    this.resourceService.getResource(resource.id, 'translations,pages').then((r) => {
+      resource.latest = r['latest-drafts-translations'];
 
-        resource.latest.forEach((translation) => {
-          this.languageService.getLanguage(translation.language.id)
-            .then((language) => {
-              translation.language = language;
-              translation.is_published = translation['is-published'];
-            });
+      resource.latest.forEach((translation) => {
```

Comment:
[`Array.prototype.forEach()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) will give you `index` as the second param in the callback. You shouldn't need to deal with initializing and incrementing `x`.